### ISSUE TITLE
List of hotfixes is empty

### ISSUE THREAD

**Issue by mauritsvanrees** [plone/plone.org#43]
Compare:

- Old: https://plone.org/security/hotfixes/
- New: https://migration.plone.org/security/hotfixes/

On the old site, this was likely a special view in `plone.app.vulnerabilities`. It combines information from the hotfix content items and three configuration registry settings: `plone.securitysupport`, `plone.versions`, `plone.activemaintenance`.

**Comment by mauritsvanrees** [comment]
There is  actually already a view that returns json with all required info.  Just call `@@hotfix_json` on the portal root (or anywhere I guess), either on the old or new site (ClassicUI).

The [code is here](https://github.com/plone/plone.app.vulnerabilities/blob/plone6/src/plone/app/vulnerabilities/browser/hotfixes.py#L67). It would probably be better to make this a restapi service, and remove the old view.
There is a lot of room for improvement in speed. The current code seems to wake up all hotfixes multiple times. No wonder that I always thought this page was slow...

I might have a look, but there are other things I need to check, and then perhaps some client work...

**Comment by mauritsvanrees** [comment]
I have made the code much faster, and created a restapi endpoint, see https://github.com/plone/plone.app.vulnerabilities/pull/12

I hope someone is willing to write a Volto view to call the `@hotfixes` endpoint for the new plone.org.

**Comment by sneridagh** [comment]
@mauritsvanrees I can't test it (I need the real data to test it), so I'd merge it and update the backend code with it, then I can start implementing the view.

**Comment by mauritsvanrees** [comment]
@sneridagh I have merged my plone.app.vulnerabilities PR so the endpoint is there.

You call `@hotfixes` on the Plone site root, no params. If it helps, I can make it available on other contexts, but it would give you the same information. As usual you get a list of dicts.

Let me call it with an optional query string to get only information about one Plone version. This returns the same information as for the full list, except of course only one dict now. I have reshuffled the keys here in this issue for clarity, and added comments.

Request: http://localhost:8080/Plone/@hotfixes?version=4.3.20

```
[
    {
        // Plone version:
        "name": "4.3.20",
        // release date of this Plone version:
        "date": "2020-08-25",
        // is version in active maintenance mode?
        "maintenance": false,
        // is version in security support?
        "security": true,
        // List of hotfixes that should be applied, possibly empty.
        "hotfixes": [
            {
                "name": "20220128",
                "release_date": "2022-01-28",
                "url": "http://localhost:8080/Plone/security/hotfix/20220128"
            },
            {
                "name": "20210518",
                "release_date": "2021-05-18",
                "url": "http://localhost:8080/Plone/security/hotfix/20210518"
            },
            {
                "name": "20151006",
                "release_date": "2015-10-06",
                "url": "http://localhost:8080/Plone/security/hotfix/20151006"
            }
        ]
    }
]
```

If both maintenance and security are false, we can hide this version. This is true for Plone 4.2 and earlier, and also for example the 5.0 release candidates. The old site has a button Show unsupported versions to toggle showing them.

And then the old page has a kind of portlet which lists all hotfixes. I guess this could just be done using the `@search` endpoint.

An alternative for now would be to copy the html of the table from https://plone.org/security/hotfixes/ and dump this in a table in a Page, editable in Volto.

**Comment by mauritsvanrees** [comment]
Okay, I have whipped up the page: https://beta.plone.org/security/hotfixes.
We want a special view for this in Volto, but I think this is okay for now and plone.org can go live with it.

**Comment by rioksane** [comment]
Looks good. I'll close the issue.

**Comment by mauritsvanrees** [comment]
The https://plone.org/security/hotfixes/ page was still private... I must have overlooked this. I have published it now.

I have changed the folder contents of the security folder, so the security announcements and the list of hotfixes is on the first row.

I noticed the announcements at https://plone.org/security/announcements were in a wrong order. The listing block seemed fine: reverse effective date. I switched it to normal (non-reversed) and then back to reversed, and then magically the order was fine: newest first.
